core/runtimes/FiberRuntime/index.js

```diff
@@ -139,20 +139,41 @@ FiberRuntime.prototype._hasDeadLock = function () {
 FiberRuntime.prototype._genInitialQueue = function () {
   var self = this;
   var queue = [];
-
+  //console.log(self._list);  //xxx
+  //console.log(Object.keys(self._list).length);
   // A process is selfstarting if its incoming ports are only connected to IIPs
+ /* ---------------------------------
   self._list.forEach(function(process) {
+	console.log(process);  //xxx
     var selfstarting = process.inports.reduce(function(currentlySelfstarting, inport) {
       return currentlySelfstarting && (inport[1].conn instanceof IIPConnection);
     }, true);
-
+    
     if(selfstarting) {
-      queue.push(process);
+        queue.push(process);
+      }
+      */
+   // -----------------------------
+    var selfstarting = true;
+    for (var key in self._list) {
```

Comment:
I'm totally cool with this.

There's a library called lodash which I would recommend we use. It allows you to use `forEach` over any kind of collection, array or object, and avoids the boilerplate of having to call `hasOwnProperty`.